### PR TITLE
Add check access database for system tables

### PR DESCRIPTION
**Pull request by zhang2014**
I'm not sure it's necessary. It helps, though, to shield messy data from multiple user scenarios.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

**Comment by filimonov** [comment]
I think system database should be generally available for all the users - as it contain system.number, system.one and other very handy tables. Only few tables can be considered as potentially unsafe: system.query_log, system.zookeeper.  System.processes can be limited to current user only for users with limited access.

File: dbms/src/Interpreters/Context.cpp [hunk 1]
```diff
@@ -618,6 +618,13 @@ void Context::checkDatabaseAccessRights(const std::string & database_name) const
     checkDatabaseAccessRightsImpl(database_name);
 }
 
+bool Context::isDatabaseAccessRights(const String & database_name) const
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/Context.cpp]: isDatabaseAccessRights -> hasDatabaseAccessRights

**Comment by alexey-milovidov** [comment]
@filimonov This pull request just filters content of system tables, that contain information about another tables. All system tables remain available, but only with information about tables that the user is allowed to see.

**Comment by alexey-milovidov** [comment]
@zhang2014 We can't filter system.processes, system.query_log, because they don't contain information about database. Even more, the query can use multiple databases. Lets just leave it as is.

**Comment by alexey-milovidov** [comment]
@zhang2014 The only missing part is integration test.

**Pull request merged**